lib/game.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require 'set'
+
+# Links board and players to create the game
+# Constraints
+# - cannot have more than 2 players
+class Game
+  attr_accessor :winner_exists
+  attr_reader :board, :current, :player_one, :player_two
+
+  def initialize(board, player_one, player_two)
+    validate_board(board)
+    validate_players(player_one, player_two)
+
+    @board = board
+    @player_one = player_one
+    @player_two = player_two
+    @current = @player_one
+    @winner_exists = false
+  end
+
+  def choose_next_player
+    @current = @current == @player_one ? @player_two : @player_one
+  end
+
+  def print_status
```

Comment:
lib folder and game logic should be free from UI, move all the put prints etc to bin and do the same where necessary.